db/src/diskdb.rs

```diff
@@ -21,7 +22,18 @@ impl RocksDB {
 
         let cfnames: Vec<_> = (0..columns).map(|c| format!("c{}", c)).collect();
         let cf_options: Vec<&str> = cfnames.iter().map(|n| n as &str).collect();
-        let db = DB::open_cf(&opts, &config.path, &cf_options).expect("Failed to open rocksdb");
+        let db = DB::open_cf(&opts, &config.path, &cf_options).unwrap_or_else(|err| {
+            warn!("Failed to open rocksdb: {}", err);
```

Comment:
Only `Corruption` error is repairable, we may need to check err message string before run repair function.

Reply:
> Only `Corruption` error is repairable, ...

Yes!

I used to be doubtful should we just [test a string to classify the error kind](https://github.com/rust-rocksdb/rust-rocksdb/blob/ea770309c8ccf99e7e2d9be0194f476bbf008346/src/lib.rs#L107-L110):
- If the error was not a `Corruption` error, repairer will do nothing. 
- If the error message was changed in upstream, we won't be noticed unless we read the ChangeLog (or source code).